src/shared_functions.rs

```diff
@@ -43,68 +43,58 @@ impl ShellState {
         ensure_directory(Path::new(&shell_state.share_dir));
         shell_state
     }
-    pub fn eval_prompt(prompt_string: &str) -> String {
-        let split_prompt: Vec<&str> = prompt_string.split("%E").collect();
-        let mut commands: Vec<ShellCommand> = Vec::new();
-        let mut maybe_sep: bool = false;
-        let mut start_record: bool = false;
-        let mut command: String = String::new();
-        for c in prompt_string.chars() {
-            if !start_record && c == '%' {
-                maybe_sep = true;
-                continue;
-            }
-            if !start_record && maybe_sep && c == 'E' {
-                maybe_sep = false;
-                start_record = true;
-                continue;
-            }
-            if start_record {
-                command.push(c);
-            }
-            if start_record && c == '%' {
-                maybe_sep = true;
-                continue;
+    pub fn eval_prompt(&mut self) -> String {
+        let mut evaled_prompt = self.prompt.clone();
```

Comment:
Before we merge the PR, I *will* figure out how to get rid of this clone.
I don't know why, but it's bugging me so much.